test/test.importance.jl

```diff
@@ -28,134 +28,165 @@ using StableRNGs: StableRNG
             @test result isa Vector{Int}
-            @test result == [33, 123, 4, 33]
+            @test result == label.(feature_importances)
         end
 
-        # Random ratio selector method
-        let result = select(feature_importances,
-                            Random(0.77, StableRNG(1))) .|> label
+        # Random sampling method with replacement
+        let result = sample(StableRNG(1),
+                            feature_importances,
+                            RandomSampler(3; replace = true)) .|> label
             @test result isa Vector{Int}
-            @test result == [33, 123, 4]
+            @test result == [4, 123, 123]
         end
 
-        # Random ratio selector in strict mode
-        @test_throws AssertionError select(feature_importances,
-                                           Random(3.1);
-                                           strict = true) .|> label
+        # Random ratio sampling method
+        let result = sample(StableRNG(2),
+                            feature_importances,
+                            RandomSampler(0.77)) .|> label
+            @test result isa Vector{Int}
+            @test result == [4, 3, 33]
+        end
+
+        # Random ratio sampling in strict mode
+        @test_throws AssertionError sample(feature_importances,
+                                           RandomSampler(3.1; strict = true))
+
+        # Random ratio sampling without strict mode
+        let result = sample(StableRNG(2),
+                            feature_importances,
+                            RandomSampler(3.1; strict = false)) .|> label
+            @test result isa Vector{Int}
+            @test result == label.(feature_importances)
+        end
 
-        # Random ratio selector without strict mode
-        let result = select(feature_importances,
-                            Random(3.1, StableRNG(1));
-                            strict = false) .|> label
+        # Random ratio sampling method with replacement
+        let result = sample(StableRNG(1),
+                            feature_importances,
+                            RandomSampler(0.77; replace = true)) .|> label
             @test result isa Vector{Int}
-            @test result == [33, 123, 4, 33]
+            @test result == [4, 123, 123]
         end
     end
 
-    @testset "Selection -- IndexBased" begin
+    @testset "ImportanceSampler" begin
         feature_importances = [4 => 12,
                                3 => 11,
                                123 => 3,
-                               33 => 1,
-                               5 => 1,
-                               7 => 1,
-                               9 => 0]
-
-        # Index based selector method
-        let result = select(feature_importances,
-                            IndexBased([3, 1, 2, 1])) .|> label
+                               33 => 1]
+
+        # Weighted random sampling method
```

Comment:
this `Weighted random sampling method` comment is not related to the `ImportanceSampler`

Reply:
But it is! ImportancesSampler is a RandomSampler which uses importances as weights.

Reply:
Ok, I get it. I think I just needed an other comment related to that. Maybe in the `src/importance.jl`, in the `ImportanceSampler` section.

Reply:
Gotcha. I added a docstring to the ImportanceSampler function to explain what goes on. HTH.